### PR TITLE
Add uretprobe example

### PR DESCRIPTION
**Pull request by ungureanuvladvictor**
This adds an example on how to use `cilium/ebpf` with `bpf2go` in order to attach a uretprobe to `readline` symbol from` /bin/bash`. The functionality of the BPF code was inspired from the example https://github.com/iovisor/bcc/blob/master/tools/bashreadline.py but adapted to work with `cilium/ebpf` loader.

Signed-off-by: Vlad Ungureanu <vladu@palantir.com>

### CODE REVIEW

File: examples/uprobe/main.go [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+
+	}
+	defer unix.Close(efd)
+
+	rd, err := ringbuffer.NewReader(objs.MapEvents, os.Getpagesize())
+	if err != nil {
+		log.Fatalf("error while createing ringbuffer reader: %v", err)
+	}
+	defer func() {
+		_ = rd.Close()
```
Comment (lmb) [inline examples/uprobe/main.go:72]: If you wait for stopped here to close the reader you can get rid of the nested go routine.

```suggestion
		<-stopper
		_ = rd.Close()
```

File: examples/uprobe/main.go [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+			default:
+			}
+			record, err := rd.Read()
+			if err != nil {
+				if ringbuffer.IsClosed(err) {
+					return
+				}
+				log.Printf("failed to read from ringbuffer: %+v\n", err)
+			}
+			binary.Read(bytes.NewBuffer(record.RawSample), binary.LittleEndian, &event)
```
Comment (lmb) [inline examples/uprobe/main.go:90]: RawSample can be empty if record.LostSamples != 0.

File: examples/headers/common.h [hunk 3]
```diff
@@ -62,4 +62,46 @@ enum {
 	BPF_F_LOCK = 4,
 };
 
+/* BPF_FUNC_perf_event_output, BPF_FUNC_perf_event_read and
+ * BPF_FUNC_perf_event_read_value flags.
+ */
+#define BPF_F_INDEX_MASK 0xffffffffULL
```
Comment (lmb) [inline examples/headers/common.h:68]: I think this is OK for now, at some point we should include `linux/bpf.h`.

**Review by lmb — CHANGES_REQUESTED** [review]

**Review by ti-mo — APPROVED** [review]
@ungureanuvladvictor Thank you, this is very useful!

**Comment by ungureanuvladvictor** [comment]
@lmb addressed your comments

**Review by lmb — APPROVED** [review]

**Comment by lmb** [comment]
Thanks!

**Pull request merged**